File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number}
     */

    //we can use a depth first search algorithm here
    maxDepth(root) {
        if (!root){
            return 0
        }

        return 1 + Math.max(this.maxDepth(root.left), this.maxDepth(root.right))
        

    }
}

File: Data Structures & Algorithms/lowest-common-ancestor-in-binary-search-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def lowestCommonAncestor(self, root: TreeNode, p: TreeNode, q: TreeNode) -> TreeNode:
        cur = root

        #cur will never be null, this is just to get the code to execute
        while cur:
            if p.val > cur.val and q.val > cur.val:
                cur = cur.right
                #search the right subtree            
            elif p.val < cur.val and q.val < cur.val:
                cur = cur.left
                #search the left subtree

            else:
                return cur
            
        
        